app/src/main/java/com/plenart/organizeme/adapters/CardListItemsAdapter.kt

```diff
@@ -6,106 +6,110 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
-import com.plenart.organizeme.activities.TaskListActivity
 import com.plenart.organizeme.databinding.ItemCardBinding
-import com.plenart.organizeme.interfaces.BoardItemClickInterface
 import com.plenart.organizeme.interfaces.CardItemClickInterface
 import com.plenart.organizeme.interfaces.MemberItemClickInterface
 import com.plenart.organizeme.models.Card
 import com.plenart.organizeme.models.SelectedMembers
+import com.plenart.organizeme.utils.CardDiffCallback
+import com.plenart.organizeme.viewModels.TaskListViewModel
 
-class CardListItemsAdapter(private val context: Context, private var list: ArrayList<Card>): RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+class CardListItemsAdapter(
+    private val context: Context,
+    private val viewModel: TaskListViewModel
```

Comment:
With this way adapter and view model are tight coupled, try to avoid that. 

The job of adapter is to handle data collection and bind it to the view. What you can do, move all logic for preparing data in ViewModel, expose it like live data, observe in fragment and then pass prepared data to adapter.

Reply:
Yes, when doing that I was aware that it's not the most optimal way but at the time I wasn't sure how to do it correctly. Thanks for the advice, will try to do it that way!